### ISSUE TITLE
vCPU shutdown with Windows 7 64

### ISSUE THREAD

**Issue by josephernest** [intel/haxm#307]
Host: Thinkpad T520, i5-2520M CPU, running Windows 7 x64
Host: QEMU 3.0.0, QEMU 3.1.0, QEMU 4.2.0
Host: intel-haxm 7.1.0 (I had to go back to this version because of https://github.com/intel/haxm/issues/211)

Guest: Windows 7 x64 installed with:

    qemu-img create -f qcow2 "windows7_cleaninstall.qcow2" 50G
    qemu-system-x86_64 -hda "windows7_cleaninstall.qcow2" -m 4096 -cdrom "fr_windows_7_ultimate_with_sp1_x64_dvd_u_677299.iso"

When running it with:

    qemu-system-x86_64 -hda "windows7_cleaninstall.qcow2" -m 4096 -accel hax

it crashes after 1 second with:

    vCPU shutdown
    vCPU shutdown
    vCPU shutdown

**Comment by josephernest** [comment]
This problem is half-solved by upgrading to `haxm-windows_v7_6_1.zip`, made possible by first installing Windows6.1-KB3033929-x64.msu.

Then there is no more:

    vCPU shutdown
    vCPU shutdown
    vCPU shutdown
    ...

but a Windows crash / blue screen with error code 

    STOP: 0x0000005D

when I run it with:

    qemu-4.2.0\qemu-system-x86_64 -hda "snapshot.qcow2" -accel hax -m 2G

On the other hand if I add `-cpu max`, it goes on with

    vCPU shutdown
    vCPU shutdown
   ...

**Comment by nevilad** [comment]
Please post hax log. Run DebugView, run qemu again, save the log in DebugView and attach here.

**Comment by josephernest** [comment]
@nevilad Nothing is displayed in DebugView, how to associate DebugView with the currently-running `qemu-system-x86_64` process?

Important note: the Windows crash / "blue screen of death" happens **inside the VM**, i.e. inside the guest, and *not* in the host. 

The host continues to work fine.

**Comment by wayne-ma** [comment]
@josephernest ,so far haxm could not support nested virtualization well and we have an internal bug tracking it.

**Comment by nevilad** [comment]
@josephernest does your debugView toolbar look like this:
![image](https://user-images.githubusercontent.com/31478632/85943762-2c92ff80-b943-11ea-842f-33b5b8da6ff9.png)
?

@wayne-ma why do you think that nested virtualization is used in this scenario?

**Comment by josephernest** [comment]
@nevilad Yes exactly.

With the guest Windows 7 x64, I have tried several hosts:

* Windows 7 host, using hax (guest crash)
* Windows 10 host, using hax (guest crash)
* Windows 10 host, using whpx (very slow)

More generally, is Win (host)  + Win (guest) not working well with QEMU?

**Comment by josephernest** [comment]
PS: 

I did the Windows 7 install *without* `--accel hax`, because I needed `--cdrom` to boot on the installation disk.

As far as I remember, you cannot use `--cdrom` and haxm at the same time, is it still true?

Could the reason of the failure be that the Windows 7 *install* was done without haxm, so, then, running the system with haxm doesn't match the system settings during the install?

**Comment by nevilad** [comment]
>More generally @nevilad, have you ever managed to run a Windows 7 x64 VM inside a Windows 7 x64 host, with haxm?

No, I've never started 64-bit windows guests under haxm. Win10 and WIn8.1 x64 guests are known to be not working. 

>As far as I remember, you cannot use -cdrom and haxm at the same time, is it still true @nevilad?

I've successfully used both options.

>Could the reason of the failure be that the Windows 7 install was done without haxm, so, then, running the system with haxm doesn't match the system settings during the install?

Hmmm, possible since cpu model is changed. I had such issues with running win10 x32 guests with changing CPU models - few runs were successfull, but after that it just didn't boot.

**Comment by josephernest** [comment]
Thank you for your answer.

@nevilad Oh really? Do you really mean "not working"?  

> Win10 and WIn8.1 x64 guests are known to be not working.

What are the main reasons Windows guests fail in a Windows host with haxm?

**Comment by nevilad** [comment]
Win10 is not working due to an unknown bug which we tried to find several times but have no success. It enters an ethernal cycle at boot.
Win8 is not working since it uses cr8 and it's support is not yet merged and released. Maybe there will be more bugs, I can't say since never ran it.

**Comment by wayne-ma** [comment]
> @josephernest does your debugView toolbar look like this:
> ![image](https://user-images.githubusercontent.com/31478632/85943762-2c92ff80-b943-11ea-842f-33b5b8da6ff9.png)
> ?
> 
> @wayne-ma why do you think that nested virtualization is used in this scenario?

Hi @nevilad , oh, my fault. I thought @josephernest wanted to run android emulator in the Win 7 guest with Win 7 host. If not, please ignore my comments. Sorry for my misunderstanding.
